### DISCUSSION TITLE
How do I put a WinFormsAvaloniaControlHost on a form?

### DISCUSSION THREAD

**Discussion by ekolis** [AvaloniaUI/Avalonia#6339] [Q&A]
I'm trying to migrate a rather complex app from WinForms to Avalonia, so I thought I'd do it piecemeal, adding a WinFormsAvaloniaControlHost to each of the Windows forms one by one and replacing the content of the forms with Avalonia views. However even after I add a reference to Avalonia.Win32 to my WinForms project, I still don't have the WinFormsAvaloniaControlHost in the toolbox to place on the form. How do I get an instance of this control so I can place it on a form? Thanks! 🙂

**Comment by jp2masa** [comment]
I think you need the [Avalonia.Win32.Interoperability](https://www.nuget.org/packages/Avalonia.Win32.Interoperability/) package.

**Reply by ekolis** [reply]
Thanks! Now I see the control, but when I try to add it to a form, I get this error:

![image](https://user-images.githubusercontent.com/2612506/127748675-4e63f526-327b-4422-8922-635e4ec93bbc.png)

This particular app is using .NET 5 and I'm running VS2019; is it only compatible with some other version of .NET/VS?

**Reply by maxkatz6** [reply]
Yes, it seems to be only .NET Framework right now. 
If somebody can adapt it to .NET 5 or Core with multitarget, it can go to next release. I think.

**Reply by ekolis** [reply]
Oh, thanks!

Yeah, I saw this warning on the package reference which explains it:

Package 'Avalonia.Win32.Interoperability 0.10.6' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework 'net5.0-windows7.0'. This package may not be fully compatible with your project.

I wonder what it would take to make it compatible...

**Reply by jp2masa** [reply]
It works for me, but you have to create it in code, something like this: [WinFormsApp1.zip](https://github.com/AvaloniaUI/Avalonia/files/6911687/WinFormsApp1.zip)

**Reply by ekolis** [reply]
Oh, thanks, that did it! Now I just need to figure out how to size the forms appropriately to their Avalonia content...
